### PR TITLE
code review changes

### PR DESCRIPTION
**Pull request by ilanDoron**

### CODE REVIEW

File: contracts/KyberIEOWrapper.sol [hunk 1]
```diff
@@ -1,7 +1,57 @@
...
+}
+
+
+contract KyberIEOWrapper is Withdrawable {
+
+    KyberNetwork kyberNetwork;
+    KyberIEO kyberIeo;
+    ERC20 constant internal ETH_TOKEN_ADDRESS = ERC20(0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee);
+
+    constructor(KyberNetwork _kyber, address _admin) public Withdrawable(_admin) {
```
Comment (yaronvel) [inline contracts/KyberIEOWrapper.sol]: we agreed network address is passed as a param every time user want to contribute

File: contracts/KyberIEOWrapper.sol [hunk 2]
```diff
@@ -1,7 +1,57 @@
...
+        uint minConversionRate, address walletId) external payable returns(uint);
+    function getExpectedRate(ERC20 src, ERC20 dest, uint srcQty) external view
+        returns (uint expectedRate, uint slippageRate);
+}
+
+
+contract KyberIEOWrapper is Withdrawable {
+
+    KyberNetwork kyberNetwork;
+    KyberIEO kyberIeo;
```
Comment (yaronvel) [inline contracts/KyberIEOWrapper.sol]: IEO? Ieo?
In any case, I think we need ieo interface here.

Comment (yaronvel) [inline contracts/KyberIEOWrapper.sol]: where do you init it? in any case should be part of the contrinute params.

File: contracts/KyberIEOWrapper.sol [hunk 3]
```diff
@@ -1,7 +1,57 @@
...
+    ERC20 constant internal ETH_TOKEN_ADDRESS = ERC20(0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee);
+
+    constructor(KyberNetwork _kyber, address _admin) public Withdrawable(_admin) {
+        require (_kyber != address(0));
+        kyberNetwork = _kyber;
+    }
+
+    event Contribution(address contributor, ERC20 token, uint amountTwei, uint tradedWei, uint change);
+    function contributeWithToken(
```
Comment (yaronvel) [inline contracts/KyberIEOWrapper.sol]: should also have max ETH (in wei) parameter.

File: contracts/KyberIEOWrapper.sol [hunk 4]
```diff
@@ -1,7 +1,57 @@
...
+        uint8 v,
+        bytes32 r,
+        bytes32 s) public returns(uint)
+    {
+        uint weiCap = kyberIeo.getContributorRemainingCap(contributor);
+        require(weiCap > 0);
+
+        uint expectedRate;
+        uint slippageRate;
+        (expectedRate, slippageRate) = kyberNetwork.getExpectedRate(token, ETH_TOKEN_ADDRESS, amountTwei);
```
Comment (yaronvel) [inline contracts/KyberIEOWrapper.sol]: why do you need it?

File: contracts/KyberIEOWrapper.sol [hunk 5]
```diff
@@ -1,7 +1,57 @@
...
+
+        require(expectedRate > 0);
+        require(token.transferFrom(contributor, this, amountTwei));
+
+        token.approve(address(kyberNetwork), amountTwei);
```
Comment (yaronvel) [inline contracts/KyberIEOWrapper.sol]: you forgot to reset it

File: contracts/KyberIEOWrapper.sol (hunk 1 above)
Comment (ilanDoron) [reply contracts/KyberIEOWrapper.sol]: don't recall.
why as a param?

File: contracts/KyberIEOWrapper.sol (hunk 4 above)
Comment (ilanDoron) [reply contracts/KyberIEOWrapper.sol]: if token isn't supported.
also to have normal min rate.

File: contracts/KyberIEOWrapper.sol (hunk 2 above)
Comment (ilanDoron) [reply contracts/KyberIEOWrapper.sol]: fixed

File: contracts/KyberIEOWrapper.sol (hunk 5 above)
Comment (ilanDoron) [reply contracts/KyberIEOWrapper.sol]: r u sure.
If I take the whole sum it zeroes. no?

**Pull request merged**